src/Faker/Provider/en_US/PhoneNumber.php

```diff
@@ -4,30 +4,118 @@
+     * @see https://en.wikipedia.org/wiki/National_conventions_for_writing_telephone_numbers#United_States.2C_Canada.2C_and_other_NANP_countries
+     */
     protected static $formats = array(
-        '+##(#)##########',
-        '+##(#)##########',
-        '0##########',
-        '0##########',
-        '###-###-####',
-        '(###)###-####',
-        '1-###-###-####',
-        '###.###.####',
-        '###-###-####',
-        '(###)###-####',
-        '1-###-###-####',
-        '###.###.####',
-        '###-###-####x###',
-        '(###)###-####x###',
-        '1-###-###-####x###',
-        '###.###.####x###',
-        '###-###-####x####',
-        '(###)###-####x####',
-        '1-###-###-####x####',
-        '###.###.####x####',
-        '###-###-####x#####',
-        '(###)###-####x#####',
-        '1-###-###-####x#####',
-        '###.###.####x#####'
+        // International format
+        '+1-{{areaCode}}-{{exchangeCode}}-####',
+        '+1 ({{areaCode}}) {{exchangeCode}}-####',
+        '+1-{{areaCode}}-{{exchangeCode}}-####',
+        '+1.{{areaCode}}.{{exchangeCode}}.####',
+        '+1{{areaCode}}{{exchangeCode}}####',
+
+        // Standard formats
+        '{{areaCode}}-{{exchangeCode}}-####',
+        '({{areaCode}}) {{exchangeCode}}-####',
+        '1-{{areaCode}}-{{exchangeCode}}-####',
+        '{{areaCode}}.{{exchangeCode}}.####',
+
+        '{{areaCode}}-{{exchangeCode}}-####',
+        '({{areaCode}}) {{exchangeCode}}-####',
+        '1-{{areaCode}}-{{exchangeCode}}-####',
+        '{{areaCode}}.{{exchangeCode}}.####',
+
+        // Extensions
+        '{{areaCode}}-{{exchangeCode}}-#### x###',
+        '({{areaCode}}) {{exchangeCode}}-#### x###',
+        '1-{{areaCode}}-{{exchangeCode}}-#### x###',
+        '{{areaCode}}.{{exchangeCode}}.#### x###',
+
+        '{{areaCode}}-{{exchangeCode}}-#### x####',
+        '({{areaCode}}) {{exchangeCode}}-#### x####',
+        '1-{{areaCode}}-{{exchangeCode}}-#### x####',
+        '{{areaCode}}.{{exchangeCode}}.#### x####',
+
+        '{{areaCode}}-{{exchangeCode}}-#### x#####',
+        '({{areaCode}}) {{exchangeCode}}-#### x#####',
+        '1-{{areaCode}}-{{exchangeCode}}-#### x#####',
+        '{{areaCode}}.{{exchangeCode}}.#### x#####'
     );
+
+    /**
+     * @see https://en.wikipedia.org/wiki/Toll-free_telephone_number#United_States
+     */
+    protected static $tollFreeAreaCodes = array(
+        800, 844, 855, 866, 877, 888
+    );
+    protected static $tollFreeFormats = array(
+        // Standard formats
+        '{{areaCode}}-{{exchangeCode}}-####',
+        '({{areaCode}}) {{exchangeCode}}-####',
+        '1-{{areaCode}}-{{exchangeCode}}-####',
+        '{{areaCode}}.{{exchangeCode}}.####',
+    );
+
+    public function tollFreePhoneNumber()
+    {
+        $areaCode = self::randomElement(static::$tollFreeAreaCodes);
+        $format = self::randomElement(static::$tollFreeFormats);
+
+        $format = str_replace('{{areaCode}}', $areaCode, $format);
```

Comment:
I guess I thought that formatter would be _too_ simple, but I guess that's not actually a problem :)